src/workload_manager.py

```diff
@@ -116,92 +103,35 @@ def is_ready(self):
         return self._container.can_connect()
 
     def _on_pebble_ready(self, _):
-        if version := self._alertmanager_version:
+        if version := self._version:
             self._unit.set_workload_version(version)
         else:
             logger.debug(
-                "Cannot set workload version at this time: could not get Alertmanager version."
+                "Cannot set workload version at this time: could not get workload version."
             )
 
     @property
-    def _alertmanager_version(self) -> Optional[str]:
-        """Returns the version of Alertmanager.
+    @abstractmethod
+    def _version(self) -> Optional[str]:
+        """Returns the version of the workload.
 
         Returns:
-            A string equal to the Alertmanager version.
+            A string equal to the workload version.
         """
-        if not self.is_ready:
-            return None
-        version_output, _ = self._container.exec([self._exe_name, "--version"]).wait_output()
-        # Output looks like this:
-        # alertmanager, version 0.23.0 (branch: HEAD, ...
-        result = re.search(r"version (\d*\.\d*\.\d*)", version_output)
-        if result is None:
-            return result
-        return result.group(1)
+        pass
 
+    @abstractmethod
     def check_config(self) -> Tuple[str, str]:
```

Comment:
I'm not sure how well this will generalize.
Why would a 'check_config' method return stderr, stdout? I'd expect it to raise if invalid, or return bool.
This is very specific to the action we use it for, and the existence of amtool. 
Maybe all our charms have such a tool, however?